enhancements/agent-installer/simplified_cluster_operations.md

```diff
@@ -0,0 +1,425 @@
+An Assisted Service pre-flight validation will ensure that the port is open.
+
+#### Upgrade
+
+A udev rule, along with a systemd service injected by MCO will detect when an
+extended RHCOS ISO is attached, an then mount it to launch a temporary registry
+to serve its content.
+After the user modified the IRI resource with the new release pullspec, the new
+OCP release image payload will be copied on each node registry using skopeo.
+
+### Risks and Mitigations
+
+* Exposing the internal registry on a node port will allow it to be reached
+  from outside the cluster. This behavior is similar to the one already
+  adopted by the Machine Config Server (MCS).
+  This is a new requirement that will need to be properly documented. In
+  addition, to minimize the risks, the registry will have to be secured
+  via a pull-secret.
+
+### Drawbacks
+
+* The extended RHCOS ISO size is pretty big, currently ~35GB with just a
+  limited set of OLM operators. Including all the OLM operators is not
+  feasible, and the current selected set may not be sufficient and/or
+  effectively required by all the users.
+* Due the size, some hardware vendors may have difficulties into loading
+  such ISO, so a more extended testing may be required.
+* The lifecycle of some OLM operators is different from the OCP one,
+  thus assembling an extended ISO may be challenging due the different release
+  timings (between OCP and OLM operators).
+* The upgrade path of some OLM operators may require to include different
+  versions of the same operator, thus impacting on the final size of the
+  extended ISO.
+* Building and releasing the extended ISO via Konflux requires a non-trivial
+  additional effort cost.
+
+## Alternatives (Not Implemented)
+
+* For building the ISO, we've explored also the usage of the [custom-coreos-disk-images](https://github.com/coreos/custom-coreos-disk-images)
+  utility, to reuse the OCP layering feature. The approach was discarded as it
+  appeared not feasible especially when dealing with such additional payload
+  size (a standalone OCP release size is usually ~20GB)
+
+* Instead of using a local registry on each node, we've also explored the
+  possibility to pre-load all the required images into an additional read-only
+  local container storage on each node. Unfortunately this approach wasn't
+  feasible, as several pieces of the OpenShift ecosystem assume that a registry
+  was always available in the environment. Removing such dependency across
+  all the possible locations appeared to be particularly complex and
+  challenging (and moreover no formal way to enforce it once removed)
+
+## Test Plan
+
+[dev-scripts](https://github.com/openshift-metal3/dev-scripts) will be enhanced
+to support testing via the extended RHCOS ISO, and new jobs will be added in the
+OpenShift CI to cover the sno/compact/ha topologies.
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
```

Comment:
Hmm, do we know how to test it e2e?
E.g. use a config-image instead of the interactive 'below-sea-level' part?